.github/workflows/js/issue.ts

```diff
@@ -0,0 +1,55 @@
+// This runs when a new issue is created, the GitHub workflow at ./issues.yml is triggered,
+// this workflow calls the Danger JS action, which transpiles + evals this file with some
+// extra runtime APIs for working with GitHub.
+
+// Docs for Danger: https://danger.systems/js/
+
+import { danger } from "danger";
+import { WebhookPayloadIssues } from "@octokit/webhooks";
+
+// Danger extracts the webhook and passes it as the first param to the default function
+export default async (webhook: WebhookPayloadIssues) => {
+  // Bail if it's anything other than the initial creation
+  if (webhook.action !== "opened") {
+    return console.log("Skipping due to not a new issue/pr");
+  }
+
+  const octokit = danger.github.api;
+  const repo = webhook.repository;
+  const author = webhook.sender;
+  const issue = webhook.issue;
+
+  // Helper for octokit API calls to this issue
+  const thisIssue = { repo: repo.name, owner: repo.owner.login, id: issue.number, issue_number: issue.number, pull_number: issue.number };
+
+  // Validate that we're on a PR
+  const pullMetadata = await octokit.pulls.get(thisIssue);
+  const isPR = pullMetadata.status === 200;
+
+  if (!isPR) {
+    return console.log("Skipping due to not being a pr, where all the current rules are based");
+  }
+
+  const isCollaboratorTeamResponse = await octokit.repos.checkCollaborator({ repo: repo.name, owner: repo.owner.login, username: author.login });
+  const accessToTypeScript = isCollaboratorTeamResponse.data.permission as string;
+
+  // This may be too lax, if so, it can be switched to checking a GitHub team
+  const isTSTeamMember = (accessToTypeScript === "admin" || accessToTypeScript === "write");
+
+  // When there isn't an assignee, assign it to yourself if you're on the team
+  if (issue.assignees.length === 0 && isTSTeamMember) {

```

Comment:
`@typescript-bot` has write-level permissions but shouldn't be assigned when it opens a PR - methinks the bot needs an exception here.